### PR TITLE
executor: control Chunk size for StreamAgg&HashAgg (#9512)

### PR DESCRIPTION
**Pull request by qw4990**
cherry-pick for #9512

### CODE REVIEW

**Comment by qw4990** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: executor/aggregate.go [hunk 1]
```diff
@@ -606,24 +601,30 @@ func (e *HashAggExec) parallelExec(ctx context.Context, chk *chunk.Chunk) error
 		e.prepare4ParallelExec(ctx)
 		e.prepared = true
 	}
-	for {
+
+	// gofail: var parallelHashAggError bool
```
Comment (lamxTyler) [inline executor/aggregate.go]: Why add this?

Comment (qw4990) [reply executor/aggregate.go]: It's a mistake to include these gofail comments.
They have been removed now.

**Comment by qw4990** [comment]
> The unit test is not picked?

Their unit tests involve some other PRs code which are not cherry-picked to v2.1;
So I didn't cherry-pick them to avoid make this PR too large just now.

Now all unit tests are cherry-picked and some involved change are also included by this PR.

**Comment by qw4990** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: executor/executor_required_rows_test.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ package executor
 import (
 	"context"
 	"fmt"
+	"github.com/pingcap/tidb/expression/aggregation"
```
[the remark this answers is not in the document]
Comment (qw4990) [reply executor/executor_required_rows_test.go]: fixed

**Comment by qw4990** [comment]
PTAL

**Comment by qw4990** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Pull request merged**